### DISCUSSION TITLE
Pygame with Beeware

### DISCUSSION THREAD

**Discussion by PyJustin** [beeware/briefcase#1164] [General]
When I Build Android for my pygame program,
it tries to download pygame but soon fails to download.

I've specified "pygame" in the Requirements of the TOML file
and it runs OK in Dev mode.

Appreciate if someone can tell me how to properly specify pygame for Beeware.

**Comment by freakboy3742** [comment]
You can't.

When you generated your project with `briefcase new`, you would have been shown the prompt:
```
What GUI toolkit do you want to use for this project?

Select one of the following:

    [1] Toga
    [2] PySide2 (does not support iOS/Android deployment)
    [3] PySide6 (does not support iOS/Android deployment)
    [4] PursuedPyBear (does not support iOS/Android deployment)
    [5] Pygame (does not support iOS/Android deployment)
    [6] None
```

As that says - Pygame isn't supported for iOS and Android development.

Briefcase can't magically make a widget toolkit support an operating system that the toolkit itself doesn't support. To the best of my knowledge, official Pygame doesn't currently have Android or iOS support.

There *is* a "renpy" Pygame subset that apparently supports Android; however, that has its own packaging and deployment tooling. It may be possible to make this subset compatible with Briefcase, but it doesn't work at present.

**Comment by PyJustin** [comment]
Oh dear ! I'm so disappointed. But thanks very much for your prompt and comprehensive reply.
